src/Routers/ExportRouter.js

```diff
@@ -0,0 +1,211 @@
+  }
+
+  handleExportProgress(req) {
+
+    const databaseController = req.config.database;
+
+    const query = {
+      masterKey: req.info.masterKey,
+      applicationId: req.info.appId
+    };
+
+    return databaseController.find(DefaultExportExportProgressCollectionName, query)
+    .then((response) => {
+      return { response };
+    });
+  }
+
+  handleExport(req) {
+
+    const databaseController = req.config.database;
+
+    const emailControllerAdapter = req.config.emailControllerAdapter;
+
+    if (!emailControllerAdapter) {
+      return Promise.reject(new Error('You have to setup a Mail Adapter.'));
+    }
+
+    const exportProgress = {
+      id: req.body.name,
+      masterKey: req.info.masterKey,
+      applicationId: req.info.appId
+    };
+
+    databaseController.create(DefaultExportExportProgressCollectionName, exportProgress)
+    .then(() => {
+      return databaseController.loadSchema({ clearCache: true});
+    })
+    .then(schemaController => schemaController.getOneSchema(req.body.name, true))
+    .then((schema) => {
+      const classNames = [ req.body.name ];
+      Object.keys(schema.fields).forEach((fieldName) => {
+        const field = schema.fields[fieldName];
+
+        if (field.type === 'Relation') {
+          classNames.push(`_Join:${fieldName}:${req.body.name}`);
+        }
+      });
+
+      const promisses = classNames.map((name) => {
+        return this.exportClass(req, { name });
+      });
+
+      return Promise.all(promisses)
+    })
+    .then((jsonFiles) => {
+
+      return new Promise((resolve) => {
+        const tmpZipFile = tmp.fileSync();
+        const tmpZipStream = fs.createWriteStream(tmpZipFile.name);
+
+        const zip  = archiver('zip');
+        zip.pipe(tmpZipStream);
+
+        jsonFiles.forEach(tmpJsonFile => {
+          zip.append(fs.readFileSync(tmpJsonFile.name), { name:  tmpJsonFile._name });
+          tmpJsonFile.removeCallback();
+        });
+
+        zip.finalize();
+
+        tmpZipStream.on('close', () => {
+
+          const buf = fs.readFileSync(tmpZipFile.name);
+          tmpZipFile.removeCallback();
+          resolve(buf);
+
+        });
+      });
+
+    })
+    .then((zippedFile) => {
+      const filesController = req.config.filesController;
+      return filesController.createFile(req.config, req.body.name, zippedFile, 'application/zip');
+    })
+    .then((fileData) => {
+
+      return emailControllerAdapter.sendMail({
+        text: `We have successfully exported your data from the class ${req.body.name}.\n
+        Please download from ${fileData.url}`,
+        link: fileData.url,
+        to: req.body.feedbackEmail,
+        subject: 'Export completed'
+      });
+    })
+    .catch((error) => {
+      return emailControllerAdapter.sendMail({
+        text: `We could not export your data to the class ${req.body.name}. Error: ${error}`,
+        to: req.body.feedbackEmail,
+        subject: 'Export failed'
+      });
+    })
+    .then(() => {
```

Comment:
should this be an always?  will this get called if an error condition happens and there's a catch?